File: c++/find-first-reoccurring-character.cpp
#include <bits/stdc++.h>
#include <string>

using namespace std;

string str = "dabcfa"; // result: a
// string str = "adbdfa"; // result: d
// string str = "dabcfe"; // result: No reoccurring characters found

map<char, bool> track;
string result = "No reoccurring characters found";

int main() {
	for( char& character : str ) {
		if( track.find(character) != track.end() ) {
			result = character;
			break;
		}

		track[character] = true;
	}

	cout << result << endl;

	return 0;
}

File: javascript/find-first-reoccurring-character.js
#!/usr/bin/node

var str = 'dabcfa'; // result: a
// var str = 'adbdfa'; // result: d
// var str = 'dabcfe'; // result: No reoccurring characters found

var track = {};
var result = 'No reoccurring characters found';

for( let character of str ) {
	if( track[character] ){
		result = character;
		break;
	}

	track[character] = true;
}

console.log(result);
